x/evm/types/proposal.go

```diff
@@ -357,3 +364,92 @@ func (mp *ManageContractMethodBlockedListProposal) FixShortAddr() {
 		}
 	}
 }
+
+type ManageSysContractAddressProposal struct {
+	Title       string `json:"title" yaml:"title"`
+	Description string `json:"description" yaml:"description"`
+	// Contract Address
+	ContractAddr sdk.AccAddress `json:"contract_address" yaml:"contract_address"`
+	IsAdded      bool           `json:"is_added" yaml:"is_added"`
+}
+
+// NewManageSysContractAddressProposal creates a new instance of NewManageSysContractAddressProposal
+func NewManageSysContractAddressProposal(title, description string, addr sdk.AccAddress, isAdded bool,
+) ManageSysContractAddressProposal {
+	return ManageSysContractAddressProposal{
+		Title:        title,
+		Description:  description,
+		ContractAddr: addr,
+		IsAdded:      isAdded,
+	}
+}
+
+// GetTitle returns title of a manage system contract address proposal object
+func (mp ManageSysContractAddressProposal) GetTitle() string {
+	return mp.Title
+}
+
+// GetDescription returns description of a manage system contract address proposal object
+func (mp ManageSysContractAddressProposal) GetDescription() string {
+	return mp.Description
+}
+
+// ProposalRoute returns route key of a manage system contract address proposal object
+func (mp ManageSysContractAddressProposal) ProposalRoute() string {
+	return RouterKey
+}
+
+// ProposalType returns type of a manage system contract address proposal object
+func (mp ManageSysContractAddressProposal) ProposalType() string {
+	return proposalTypeManageSysContractAddress
+}
+
+// ValidateBasic validates a manage system contract address proposal
+func (mp ManageSysContractAddressProposal) ValidateBasic() sdk.Error {
+	//will delete it after upgrade venus3
+	if global.GetGlobalHeight() > 0 && !types.HigherThanVenus3(global.GetGlobalHeight()) {
+		return govtypes.ErrInvalidProposalContent("not support system contract address proposal")
+	}
+
+	if len(strings.TrimSpace(mp.Title)) == 0 {
+		return govtypes.ErrInvalidProposalContent("title is required")
+	}
+
+	if len(mp.Title) > govtypes.MaxTitleLength {
+		return govtypes.ErrInvalidProposalContent("title length is longer than the maximum title length")
+	}
+
+	if len(mp.Description) == 0 {
+		return govtypes.ErrInvalidProposalContent("description is required")
+	}
+
+	if len(mp.Description) > govtypes.MaxDescriptionLength {
+		return govtypes.ErrInvalidProposalContent("description length is longer than the maximum description length")
+	}
+
+	if mp.ProposalType() != proposalTypeManageSysContractAddress {
+		return govtypes.ErrInvalidProposalType(mp.ProposalType())
+	}
+
+	if mp.IsAdded && mp.ContractAddr.Empty() {
```

Comment:
mp.Contract can not be empty, and not sdk.IsWasmAddress(mp.Contract)

Reply:
> mp.Contract can not be empty, and not sdk.IsWasmAddress(mp.Contract)

yes, if IsAdded is false, ContractAddr can be empty